KMeans.java

```diff
@@ -0,0 +1,126 @@
+import java.io.*;
+import java.util.*;
+import java.lang.Math; 
+
+public class KMeans {
+
+	private Vector midpoint (List<Vector> listofvectors) { //computes + returns a midpoint Vector from a given list of Vectors
+		
+		List<Double> mp_coordinates = new ArrayList<Double>();
+		double number_of_vectors = listofvectors.size();
+		double length_of_vector = listofvectors.get(0).numbers.size();	
```

Comment:
use `lowerCamelCase` for local variables (here and everywhere else).  And this should be `vectorLength`.